### DISCUSSION TITLE
Fix inconsistencies between Application path helpers

### DISCUSSION THREAD

**Discussion by jnoordsij** [laravel/framework#45870] [Ideas] [answered]
The `Application` class has various path-related functions, like `configPath` and `databasePath`. A large number of them are also part of the `Application` contract.

However, I noticed that the `publicPath` function differs from the others in that 
1. it does not accept an argument
2. there is a version in `helpers.php` that  has its very own logic, instead of calling the `publicPath` helper on the `Application` class, which actually does accept an argument
3. it is not present in the contract class

I think it would make sense to move all the logic to the `Application` class and also add it to the contract, but I'm not sure if there are any particular reasons to why it is currently implemented in a way different from the other functions.

**Comment by parallels999** [comment]
Hi, try to make a PR to 10.x branch

**Comment by jnoordsij** [comment, marked as answer]
Created #45968 and #45969 to resolve this.
